Robot Return to Origin

OlegCherep opened the pull request:
Вернуть True, если робот, который может ходить вверх, вниз, влево и вправо вернётся в точку (0, 0) и False в противном случае.
Ссылка: https://leetcode.com/problems/robot-return-to-origin/

OlegCherep committed:
Robot Return to Origin

Вернуть True, если робот, который может ходить вверх, вниз, влево и впрво вернётся в точку (0, 0) и False в пртивном случае.

vitkarpov commented on Robot Return to Origin.py:
```diff
@@ -0,0 +1,8 @@
+    def judgeCircle(self, moves: str) -> bool:
+        m = {"U": [],"D": [],"L": [],"R": []}
+        for i in range(len(moves)):
+            m[moves[i]].append(i)
```
А точно нужно запоминать индексы, как они потом используются? Кажется, что только чтобы взять длину этого массива, соответственно возникает вопрос не проще ли использовать просто одно число вместо массива.

OlegCherep replied:
Да, можно каждую букву обозначить ключом, а в качестве значения можно взять 0 и прибавлять 1, когда встречается та или иная буква.

vitkarpov commented on Robot Return to Origin.py:
```diff
@@ -0,0 +1,8 @@
+        m = {"U": [],"D": [],"L": [],"R": []}
+        for i in range(len(moves)):
+            m[moves[i]].append(i)
+        if len(m['U']) == len(m['D']) and (len(m['R']) == len(m['L'])):
+            return True
+        return False
```
можно было бы вернуть результат вычисления `len(m['U']) == len(m['D']) and (len(m['R']) == len(m['L']))`, ведь это и есть boolean. Сейчас это читается пример как "если true, то верни true, а иначе верни false"

OlegCherep replied:
Да, точно. Можно было просто return (m['U'] == m['D']) and (m['R'] == m['L']) написать и всё

OlegCherep committed:
Update Robot Return to Origin.py

Изменил код, присвоив в словаре каждой букве счётчик и написал более грамотный вывод результата вычислений.

vitkarpov reviewed (approved):
Ещё можно просто два счетчика: горизонталь и вертикаль. Если в итоге баланс сошёлся в ноль, то робот вернулся в исходную точку.
